pandas/core/generic.py

```diff
@@ -5213,6 +5213,15 @@ def _make_stat_function(name, name1, name2, axis_descr, desc, f):
     @Appender(_num_doc)
     def stat_func(self, axis=None, skipna=None, level=None, numeric_only=None,
                   **kwargs):
+        compat_args = ['out', 'dtype']
+        list(map(kwargs.__delitem__, filter(
+                kwargs.__contains__, compat_args)))
+        if kwargs:
```

Comment:
can you put this in a helper function (you can pass ``kwargs`` and parameterize the ``compat_args`` if you want)